src/Symfony/Bundle/FrameworkBundle/Test/HttpClientAssertionsTrait.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+/*
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ */
+
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Component\HttpClient\DataCollector\HttpClientDataCollector;
+
+trait HttpClientAssertionsTrait
+{
+    public static function assertHttpClientHasBeenCalled(KernelBrowser $client, string $httpClientId, array $urls): void
+    {
+        if (($profile = $client->getProfile()) === false) {
+            static::fail('The Profiler must be enabled for the current request. Please ensure you have called $client->enableProfiler() before making the request.');
+        }
+
+        /** @var HttpClientDataCollector $httpClientDataCollector */
+        $httpClientDataCollector = $profile->getCollector('http_client');
+
+        foreach ($urls as $i => $url) {
+            self::assertSame($url, $httpClientDataCollector->getClients()[$httpClientId]['traces'][$i]['info']['url']);
```

Comment:
This will trigger a notice if the trace `$i` does not exist, which is a bad way to fail the test if the expected call was not done.

Reply:
Yes, sure. I will add a check on the sizes of both arrays, and use array_values to ensure keys are reset.

Reply:
I've abandoned the idea to respect the order. It could lead to lots of false positive, as you said "after all, the client itself might run requests concurrently depending on how you triggered them".